src/helpViewer/helpProvider.ts

```diff
@@ -270,33 +268,54 @@ export class AliasProvider {
     }
 
     // call R script `getAliases.R` and parse the output
-    private getAliasesFromR(): undefined | AllPackageAliases {
-
-        // get from R
-		const lim = '---vsc---'; // must match the lim used in R!
-		const re = new RegExp(`^.*?${lim}(.*)${lim}.*$`, 'ms');
-        const tempDir = fs.mkdtempSync(path.join(os.tmpdir(), 'vscode-R-aliases'));
-        const tempFile = path.join(tempDir, 'aliases.json');
-        const cmd = `"${this.rPath}" --silent --no-save --no-restore --slave -f "${this.rScriptFile}" > "${tempFile}"`;
-
-        let allPackageAliases: undefined | AllPackageAliases = undefined;
-        try{
-            // execute R script 'getAliases.R'
-            // aliases will be written to tempFile
-            cp.execSync(cmd, { cwd: this.cwd });
-
-            // read and parse aliases
-            const txt = fs.readFileSync(tempFile, 'utf-8');
-            const json = txt.replace(re, '$1');
-            if(json){
-                allPackageAliases = <{[key: string]: PackageAliases}> JSON.parse(json) || {};
+    private async getAliasesFromR(): Promise<undefined | AllPackageAliases> {
+        const lim = '---vsc---';
+        const options: cp.ExecSyncOptionsWithStringEncoding = {
+            cwd: this.cwd,
+            encoding: 'utf-8',
+            env: {
+                ...process.env,
+                VSCR_LIM: lim
             }
-        } catch(e: unknown){
-            console.log(e);
-            void window.showErrorMessage((<{message: string}>e).message);
-        } finally {
-            fs.rmdirSync(tempDir, {recursive: true});
-        }
-        return allPackageAliases;
+        };
+
+        const args = [
+            '--silent',
+            '--slave',
+            '--no-save',
+            '--no-restore',
+            '-f',
+            this.rScriptFile
+        ];
+
+        return new Promise((resolve) => {
+            try {
+                let str = '';
+                const childProcess = spawn(this.rPath, args, options);
+                childProcess.stdout?.on('data', (chunk: Buffer) => {
+                    str += chunk.toString();
+                });
+                childProcess.on('exit', (code, signal) => {
+                    let result: AllPackageAliases | undefined = undefined;
+                    if (code === 0) {
+                        const re = new RegExp(`${lim}(.*)${lim}`, 'ms');
+                        const match = re.exec(str);
+                        if (match.length === 2) {
+                            const json = match[1];
+                            result = <{ [key: string]: PackageAliases }>JSON.parse(json) || {};
```

Comment:
The type could be expressed as `AllPackageAliases` as above

Reply:
Yes thanks!